### PR TITLE
Rule R4005 XmsEnumNameUnique

### PR DESCRIPTION
**Pull request by jianyexi**

### CODE REVIEW

File: src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export const XmsEnumNameUnique: string = "XmsEnumNameUnique";
+import { nodes } from "jsonpath";
+
+var enumList: string[] = []
+
+rules.push({
+  id: "R4005",
+  name: XmsEnumNameUnique,
+  severity: "warning",
+  category: "ARMViolation",
```
Comment (lmazuel) [inline src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts]: It's not just ARM but everything autorest related

Comment (jianyexi) [reply src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts]: for this rule ,  it  just impact code generation

File: src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      )) {
+        if (section.value.name) {
+          enumList.push(section.value.name);
+        }
+      }
+      const caseInsensitiveSet = new Set<string>();
+      if (
+        enumList.some(value => {
+          if (caseInsensitiveSet.has(value.toLowerCase())) {
+            console.log("return true")
```
Comment (ruowan) [inline src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts]: Do we need the log?

File: src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts (hunk 1 above)
Comment (lmazuel) [reply src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts]: It impacts code generation cross-context, not only for ARM

**Pull request merged**